### ISSUE TITLE
counter labels flexibility

### ISSUE THREAD

**Issue by NikolaBorisov** [deadtrickster/prometheus.erl#114]
Hello,

It looks like the prometheus_counter:inc and prometheus_gauge:set require you to pass just the LabelValues. However I would like to be able to have metrics where some labels are sometimes there and sometimes not. 

for example I would like to be able to do something like this:
stat:count("metric1", [{foo, bar}, {x, y}], 1)
stat:count("metric1", [{foo, bar}], 1)
stat:count("metric1", [{x, y}, {foo, bar}], 1)

Basically my labels are not always the same number and not always in the same order. I think this is generally supported by Prometheus but not really possible with prometheus.erl api right now...

**Comment by NikolaBorisov** [comment]
I don't think we want to fix this.
